File: asp.net-projects/Maps_Project/Maps_Project.Web/Modules/ServerTypes/Trip/BusinessTripDetailColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { BusinessTripDetailRow } from "./BusinessTripDetailRow";

export interface BusinessTripDetailColumns {
    Id: Column<BusinessTripDetailRow>;
    BusinessTripTitle: Column<BusinessTripDetailRow>;
    Coordinate: Column<BusinessTripDetailRow>;
}

export class BusinessTripDetailColumns extends ColumnsBase<BusinessTripDetailRow> {
    static readonly columnsKey = 'Trip.BusinessTripDetail';
    static readonly Fields = fieldsProxy<BusinessTripDetailColumns>();
}

File: asp.net-projects/Maps_Project/Maps_Project.Web/Modules/Trip/BusinessTripDetail/BusinessTripDetailColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;
using System.IO;

namespace Maps_Project.Trip.Columns;

[ColumnsScript("Trip.BusinessTripDetail")]
[BasedOnRow(typeof(BusinessTripDetailRow), CheckNames = true)]
public class BusinessTripDetailColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string BusinessTripTitle { get; set; }
    public string Coordinate { get; set; }
}